### PR TITLE
[DebugBundle] Fix dump() output in API / No-Toolbar context

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

With this PR, dumps are written as plain text when the main content-type is not HTML.
With HTML, dumps are inlined when the web debug toolbar is not enabled.

### CODE REVIEW

**Comment by fabpot** [comment]
:+1:

File: src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php [hunk 1]
```diff
@@ -140,6 +174,7 @@ public function serialize()
         $this->data = array();
         $this->dataCount = 0;
         $this->isCollected = true;
+        $this->dumper = null;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php]: I just added this new line to make the service reusable, aka state less after use

**Comment by stof** [comment]
this looks weird to me. It means that your debugging output will not break the content of the response (a json response would not be valid json anymore for instance).
This seems a bad idea IMO.

And for instance, for AJAX requests, the link to the profile is available in the toolbar of the main page even though the toolbar is not injected in the ajax response. So the profile is accessible in such case.

**Comment by nicolas-grekas** [comment]
This is just what happens when you use var_dump(), with a better experience: currently, you either have no output at all, which is a huge WTF, or an HTML blob, regardless of the actual content-type.

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by nihilnovi** [comment]
Noticed this change today when debugging an api. Used to be that you could call "dump" and check the profiler without having a broken api response, with this change that is completely broken now. Is there any way to invoke the old behaviour?

**Comment by stof** [comment]
@nihilnovi please open an issue to discuss it. Reporting issues by commenting on merged PRs is not working well because anyone not reading Github notifications would miss the discussion given they will only check the opened issues and PRs

**Comment by nihilnovi** [comment]
@stof thanks, you are correct,  I will do so

**Pull request merged**